src/vma/sock/sockinfo_tcp.cpp

```diff
@@ -3041,7 +3041,7 @@ int sockinfo_tcp::wait_for_conn_ready()
 		 * therefore in this case the m_conn_state will not be changed only
 		 * m_sock_state
 		 */
-		if (rx_wait(poll_count, m_b_blocking) < 0) {
+		if (rx_wait(poll_count, m_b_blocking) < 0 || unlikely(g_b_exit)) {
```

Comment:
Please consider setting errno as done in other places
see:
https://github.com/Mellanox/libvma/blob/master/src/vma/sock/sockinfo_tcp.cpp#L626-L628
https://github.com/Mellanox/libvma/blob/master/src/vma/sock/sockinfo_tcp.cpp#L793-L797